src/kvasir/mpl/algorithm/all.hpp

```diff
@@ -4,26 +4,37 @@
 //          http://www.boost.org/LICENSE_1_0.txt)
 #pragma once
 
-#include "remove_if.hpp"
-#include "../sequence/size.hpp"
-#include "../functional/compose.hpp"
-#include "../utility/invert.hpp"
+#include "../functional/call.hpp"
+#include "../functional/bind.hpp"
+#include "../algorithm/find_if.hpp"
 
 namespace kvasir {
 	namespace mpl {
-		namespace impl {
-			/// generic implementation for any list type
-			template <template <typename...> class Cond, typename List>
-			struct all_impl {
-				constexpr operator bool() const {
-					return size_impl<remove_if_impl<Cond, List>>{} == 0;
-				}
-			};
-		};
+		namespace c {
+			namespace detail {
+				struct nothing_found {
+					template<typename...Ts>
+					using f = std::integral_constant<bool, (sizeof...(Ts) == 0)>;
+				};
+				template<typename F>
+				struct not_ {
+					template<typename T>
+					using f = std::bool_constant<(!F::template f<T>::value)>;
+				};
+				template<template<typename...> class F>
+				struct not_<bind<F>> {
+					template<typename T>
+					using f = std::bool_constant<(!F<T>::value)>;
+				};
+			}
 
-		/// filter elements from a list
-		/// takes a lambda that should return a type convertible to bool
+			template<typename F>
+			struct all : find_if<detail::not_<F>, detail::nothing_found> {};
```

Comment:
not sure what you mean here, I'm inheriting from another continuation so that its alias is called directly